### PR TITLE
change node dirtyFlag name & add interface

### PR DESCRIPTION
**Pull request by LinYunMo**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/scene-graph/node.ts [hunk 1]
```diff
@@ -2467,6 +2467,14 @@ export class Node extends CCObject implements ISchedulable, CustomSerializable {
         }
     }
 
+    /**
+     * @en Does the world transform information of this node need to be updated?
+     * @zh 这个节点的空间变换信息是否需要更新？
+     */
+    public isTransformDirty () {
+        return this._transformFlags | TransformBit.NONE;
```
Comment (minggo) [inline cocos/scene-graph/node.ts]: I think should use `^|` here.

File: native/cocos/core/scene-graph/Node.h [hunk 2]
```diff
@@ -651,7 +650,7 @@ class Node : public CCObject {
     // NOTE: TypeArray created in node.jsb.ts _ctor should have the same memory layout
     uint32_t _eventMask{0};                                             // Uint32: 0
     uint32_t _layer{static_cast<uint32_t>(Layers::LayerList::DEFAULT)}; // Uint32: 1
-    uint32_t _dirtyFlag{0};                                             // Uint32: 2
+    uint32_t _transformFlags{0};                                             // Uint32: 2
```
Comment (minggo) [inline native/cocos/core/scene-graph/Node.h]: Please fix the code format.

File: native/cocos/core/scene-graph/Node.cpp [hunk 3]
```diff
@@ -520,9 +520,9 @@ Mat4 Node::getWorldRT() {
 void Node::invalidateChildren(TransformBit dirtyBit) { // NOLINT(misc-no-recursion)
     auto curDirtyBit{static_cast<uint32_t>(dirtyBit)};
     const uint32_t hasChangedFlags = getChangedFlags();
-    const uint32_t dirtyFlags = getDirtyFlag();
+    const uint32_t dirtyFlags = _transformFlags;
```
Comment (dumganhar) [inline native/cocos/core/scene-graph/Node.cpp]: The variable name `dirtyFlags` should be renamed to `transformFlags` either.

File: native/tools/swig-config/scene.i [hunk 4]
```diff
@@ -137,6 +137,7 @@ using namespace cc;
 %ignore cc::Node::getWorldMatrix;
 %ignore cc::Node::getWorldRS;
 %ignore cc::Node::getWorldRT;
+%ignore cc::Node::isTransformDirty;
```
Comment (dumganhar) [inline native/tools/swig-config/scene.i:140]: Why ignores it?

File: native/cocos/core/scene-graph/Node.h [hunk 5]
```diff
@@ -471,8 +471,7 @@ class Node : public CCObject {
         _hasChangedFlags = value;
     }
 
-    inline void setDirtyFlag(uint32_t value) { _dirtyFlag = value; }
-    inline uint32_t getDirtyFlag() const { return _dirtyFlag; }
+    inline bool isTransformDirty() const { return _transformFlags != static_cast<uint32_t>(TransformBit::TRS); }
```
Comment (minggo) [inline native/cocos/core/scene-graph/Node.h]: I think the logic is not correct. For example, `TransformBit::POSITION != TransformBit::TRS`, but it is dirty. Can use `_transformFlags  ^| TransformBit::None`, or `_transformFlags  != TransformBit::None`.

File: native/cocos/core/scene-graph/Node.cpp [hunk 6]
```diff
@@ -495,7 +495,7 @@ void Node::updateWorldTransformRecursive(uint32_t &dirtyBits) { // NOLINT(misc-n
             }
         }
     }
-    setDirtyFlag(static_cast<uint32_t>(TransformBit::NONE));
+    _transformFlags = (static_cast<uint32_t>(TransformBit::NONE));
```
Comment (minggo) [inline native/cocos/core/scene-graph/Node.cpp:498]: It seems `setDirtyFlag()` is not need? Can delete it?

Comment (LinYunMo) [reply native/cocos/core/scene-graph/Node.cpp:498]: Already removed

File: native/cocos/core/scene-graph/Node.h (hunk 5 above)
Comment (LinYunMo) [reply native/cocos/core/scene-graph/Node.h]: fixed

File: native/tools/swig-config/scene.i (hunk 4 above)
Comment (LinYunMo) [reply native/tools/swig-config/scene.i:140]: Added code to node.jsb.ts

File: native/cocos/core/scene-graph/Node.cpp (hunk 3 above)
Comment (LinYunMo) [reply native/cocos/core/scene-graph/Node.cpp]: fixed

File: native/cocos/core/scene-graph/Node.h (hunk 2 above)
Comment (LinYunMo) [reply native/cocos/core/scene-graph/Node.h]: fixed

File: cocos/scene-graph/node.ts (hunk 1 above)
Comment (LinYunMo) [reply cocos/scene-graph/node.ts]: change it to  !==

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Comment by LinYunMo** [comment]
@cocos-robot run test cases

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**